### PR TITLE
Use default parameters where possible in examples

### LINKED ISSUE #346: Use default parameters in examples
Many examples still define all parameters since they were created before #332 was merged. We should update the examples to only use custom settings where necessary.

### PR DESCRIPTION
**Pull request by jakobj**
As the title suggests. This should make all examples easier to understand.

fixes #346

### CODE REVIEW

File: examples/example_hurdles.py [hunk 1]
```diff
@@ -86,62 +81,34 @@ def objective_two(individual):
...
-ea_params = {
-    "n_offsprings": 4,
-    "mutation_rate": 0.03,
-    "n_processes": 1,
-    "hurdle_percentile": [0.5, 0.0],
-}
+# Next, we set up the evolutionary search. We first define the parameters of the
+# evolutionary algorithm. We define the upper percentile of individuals which
+# are evaluated on the (n+1)th objective by a list of numbers between 0 and 1
+ea_params = {"hurdle_percentile": [0.75, 0.0]}
```
Comment (HenrikMettler) [inline examples/example_hurdles.py:102]: Just out of curiosity: Why the percentile change?

File: examples/example_hurdles.py [hunk 2]
```diff
@@ -193,4 +161,4 @@ def recording_callback(pop):
 ax_function.set_ylabel(r"$f(x)$")
 ax_function.set_xlabel(r"$x$")
 
-fig.savefig("example_minimal.pdf", dpi=300)
+fig.savefig("example_hurdles.pdf", dpi=300)
```
Comment (HenrikMettler) [inline examples/example_hurdles.py:164]: nice catch :)

File: examples/example_parametrized_nodes.py [hunk 3]
```diff
@@ -96,23 +96,18 @@ def objective(individual, seed):
 
 
 # %%
-# Next, we define the parameters for the population, the genome of
+# Next, we define the parameters for the genome of
 # individuals, the evolutionary algorithm, and the local search. Note
 # that we add the custom node defined above as a primitive.
 
-population_params = {"n_parents": 1, "seed": 818821}
+seed = 818821
```
Comment (HenrikMettler) [inline examples/example_parametrized_nodes.py:103]: `seed = 1234`

File: examples/example_piecewise_target_function.py [hunk 4]
```diff
@@ -72,55 +72,35 @@ def objective(individual, rng):
...
 # Next, we set up the evolutionary search. First, we define the parameters for
-# the population, the genomes of individuals, and the evolutionary
-# algorithm.
-population_params = {
-    "n_parents": 1,
-    "seed": 8188211,
-}
+# the genomes of individuals, and the evolutionary algorithm.
+
+seed = 818821
```
Comment (HenrikMettler) [inline examples/example_piecewise_target_function.py:77]: `seed = 1234`

File: examples/example_reorder.py [hunk 5]
```diff
@@ -50,59 +50,37 @@ def objective(individual):
...
-    "n_offsprings": 4,
-    "mutation_rate": 0.03,
-    "n_processes": 2,
-    "reorder_genome": True,
-}
+# Next, we set up the evolutionary search. We first define the parameters for
+# the genome of individuals, and two evolutionary algorithms without (default)
+# and with genome reordering.
+
+seed = 818821
```
Comment (HenrikMettler) [inline examples/example_reorder.py:71]: `seed = 1234`

**Review by HenrikMettler — CHANGES_REQUESTED** [review]
Thanks a lot for taking care of this tedious task! I made some small observations. Also I didn't run the examples, I assume you did, to make sure they still work?

File: examples/example_hurdles.py (hunk 2 above)
Comment (jakobj) [reply examples/example_hurdles.py:164]: yeah, we do like copy and paste :D

File: examples/example_hurdles.py (hunk 1 above)
Comment (jakobj) [reply examples/example_hurdles.py:102]: just to make the runtime difference a bit more obvious. i thought i'd sneak it in with this PR ;)

File: examples/example_parametrized_nodes.py (hunk 3 above)
Comment (jakobj) [reply examples/example_parametrized_nodes.py:103]: you mean bc it looks too hand picked? ;) should we change all seeds in the examples to 1234 then?

in general i think we can not avoid some handpicking just bc we want the examples to finish in reasonable time. a bad seed can prevent the search from finding a good solution. :(

Comment (HenrikMettler) [reply examples/example_parametrized_nodes.py:103]: I just copied it from above, where you changed It in an example, however if possible we can just use the same everywhere?

Comment (jakobj) [reply examples/example_parametrized_nodes.py]: yes, it seems like we can use the same everywhere. just pushed a new commit where the seeds are all identical (except a few cases where we want to use multiple different seeds in the same script).

**Comment by jakobj** [comment]
> Thanks a lot for taking care of this tedious task! I made some small observations. Also I didn't run the examples, I assume you did, to make sure they still work?

thanks for the review! addressed all your comments. i think i tried all examples, but in any case, CI should catch those that don't work. :crossed_fingers:

**Review by HenrikMettler — APPROVED** [review]
LG! Thanks for taking care of this! 🚀

**Comment by HenrikMettler** [comment]
ah, before I merge, could you squash the two commits? thx

**Comment by jakobj** [comment]
thx for the review! squashed and pushed, let's see whether CI is happy.

**Pull request merged**